packages/react-components/react-tree/src/components/TreeItem/useTreeItem.tsx

```diff
@@ -89,7 +93,7 @@ export function useTreeItem_unstable<Value = string>(
     requestOpenChange({
       event,
       value,
-      open: isLeaf ? open : !open,
+      open: isBranch ? !open : open,
```

Comment:
There's a lot of conditional code paths here that depends on `isBranch`,  even if you're not considering making separate components for the item types it would still make sense to do something similar to `useSubtree` and `useRootTree`

Reply:
I was considering that actually, might be a simplification for the future